### ISSUE TITLE
upload only image format 

### ISSUE THREAD

**Issue by nikhilsmsc** [dfa1234/ngx-image-compress#13]
while uploading it is taking all format in popup window , i want to show only image format like jpeg, png like formats can you help me in this

**Comment by AncaorDev** [comment]
> while uploading it is taking all format in popup window , i want to show only image format like jpeg, png like formats can you help me in this

that does not depend on the plugin, that's what the browser does, use accept = "image/jpeg, image/png"

example 

`<input type="file" style="display:none" #fileTxt (change)="changeImage($event)" accept="image/jpeg, image/png" multiple>`

**Comment by nikhilsmsc** [comment]
Thanks for reply ,
But in changeimage event we don't have orientation parameter , what i need to pass ?
and one more doubt can i compress image in file format ? or else we need to use only base64 format only

**Comment by dfa1234** [comment]
It's very simple
`image` is the base64 string in the callback here:
```typescript
this.imageCompress.uploadFile().then(({image, orientation}) => {
      console`.log(image)
      // it's like "data:image/png;base64,iVBORw0KGg...."
      let isPNG:boolean = image.includes('image/png')
})
```
You don't need a input type file.
It's already created and handled by this function.
Just use a button with a `(click)` calling this function for uploading

**Comment by PunkHaz4rd** [comment]
Is there a handler to force types in the selection window ? Because AFAIK for mobile browsers it doesn't work.

**Comment by PunkHaz4rd** [comment]
Nevermind, I used the regular file input, got the base64 from the FileReader then proceeded to compress it with ngx-compress-img.

**Comment by dfa1234** [comment]
If you can post your custom input (html only) I will adapt for mobile.

**Comment by svralkar30** [comment]
To include all the format of the image. use the following simple solution

```
this.imageCompress.uploadFile().then(({image, orientation}) => {
   let type: any = image.match(/[^:]\w+\/[\w-+\d.]+(?=;|,)/)[0];
      let filetype = type.split('/');
      if (filetype[0] == 'image') {
this.imageCompress.compressFile(image, orientation, 50, 50).then(
          result => {
            this.uploadFile(result, filetype[1])
          });
}
})
```

it will only accept the image type file

```

**Comment by byronigoe** [comment]
This is resolved by #17.

**Comment by dfa1234** [comment]
Thank a lot @byronigoe, I'm closing this
